packages/common/src/utils/analytics/amplitude.ts

```diff
@@ -1,129 +1,105 @@
-  } catch (e) {
-    console.warn(e);
-  }
-};
-
-const getAmplitude = async (): Promise<any | false> => {
-  if (process.env.NODE_ENV !== 'production') {
-    return false;
-  }
-
-  for (let i = 0; i < 10; i++) {
-    if (
-      typeof global.amplitude !== 'undefined' &&
-      global.amplitude.getInstance()._storageSuffix
-    ) {
-      return global.amplitude;
-    }
-
-    // eslint-disable-next-line no-await-in-loop
-    await delay(1000);
-  }
-
-  return false;
+export const init = (apiKey: string) => {
+  amplitude.init(apiKey, undefined, {
+    serverUrl: 'https://stats.codesandbox.io/2/httpapi',
+    appVersion: VERSION,
+    flushIntervalMillis: 5000,
+    defaultTracking: {
+      attribution: false,
+      pageViews: false, // We have custom logic for pageview tracking
+    },
+  });
 };
 
 export const identify = async (key: string, value: any) => {
-  const amplitude = await getAmplitude();
-  if (amplitude) {
-    const identity = new amplitude.Identify();
-    identity.set(key, value);
-    amplitude.identify(identity);
-    debug('[Amplitude] Identifying', key, value);
-  } else {
+  if (!amplitude) {
     debug('[Amplitude] NOT identifying because Amplitude is not loaded');
   }
+
+  const identity = new amplitude.Identify();
+  identity.set(key, value);
+  amplitude.identify(identity);
+  debug('[Amplitude] Identifying', key, value);
 };
 
 export const identifyOnce = async (key: string, value: any) => {
-  const amplitude = await getAmplitude();
-  if (amplitude) {
-    const identity = new amplitude.Identify();
-    identity.setOnce(key, value);
-    amplitude.identify(identity);
-    debug('[Amplitude] Identifying', key, value);
-  } else {
+  if (!amplitude) {
     debug('[Amplitude] NOT identifying because Amplitude is not loaded');
   }
+
+  const identity = new amplitude.Identify();
+  identity.setOnce(key, value);
+  amplitude.identify(identity);
+  debug('[Amplitude] Identifying', key, value);
 };
 
 export const setUserId = async (userId: string) => {
-  const amplitude = await getAmplitude();
-  if (amplitude) {
-    debug('[Amplitude] Setting User ID', userId);
-    identify('userId', userId);
-
-    amplitude.getInstance().setUserId(userId);
-  } else {
+  if (!amplitude) {
     debug('[Amplitude] NOT setting userid because Amplitude is not loaded');
   }
+
+  identify('userId', userId);
+  amplitude.setUserId(userId);
+  debug('[Amplitude] Setting User ID', userId);
 };
 
 export const resetUserId = async () => {
-  const amplitude = await getAmplitude();
-  if (amplitude) {
-    debug('[Amplitude] Resetting User ID');
-    identify('userId', null);
-
-    if (
-      amplitude.getInstance().options &&
-      amplitude.getInstance().options.userId
-    ) {
-      amplitude.getInstance().setUserId(null);
-      amplitude.getInstance().regenerateDeviceId();
-    }
-  } else {
+  if (!amplitude) {
     debug('[Amplitude] NOT resetting user id because Amplitude is not loaded');
   }
+
+  amplitude.reset();
```

Comment:
Are we sure that we reset the user id fully here? Looking at the previous code (with the explicit null setting), I have a feeling that there's a bigger story here and maybe it requires the clear null setting.

Reply:
I was looking at the docs for the browser SDK and the reset does both user reset and device id: https://www.docs.developers.amplitude.com/data/sdks/browser-2/#reset-when-user-logs-out